test/python/test_operation.py

```diff
@@ -37,7 +37,20 @@
 def _tuple_to_string(t):
     lt = [str(x) for x in t]
     return '(' + ', '.join(lt) + ')'
-
+def eval_numerical_gradient(f, x,y):
```

Comment:
This function calculates the gradient of f at x, not y. Please be careful when you use it.